homeassistant/components/weather/__init__.py

```diff
@@ -153,13 +404,40 @@ def humidity(self) -> float | None:
 
     @property
     def wind_speed(self) -> float | None:
-        """Return the wind speed in native units."""
+        """Return the wind_speed for backward compatibility.
+
+        Should not be set by integrations.
+        """
         return self._attr_wind_speed
 
     @property
-    def wind_speed_unit(self) -> str | None:
+    def native_wind_speed(self) -> float | None:
+        """Return the wind speed in native units."""
+        return self._attr_native_wind_speed
```

Comment:
The deprecated `wind_speed` reports the native pressure, not the display pressure so we should return it here:
```suggestion
        if (pressure := self.wind_speed) is not None:
            return wind_speed

        return self._attr_native_wind_speed
```